packages/file_selector/file_selector_android/lib/file_selector_android.dart

```diff
@@ -0,0 +1,148 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:file_selector_platform_interface/file_selector_platform_interface.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/services.dart';
+
+const MethodChannel _channel =
```

Comment:
This should be using Pigeon, not method channels directly, as we're generally using Pigeon for new code when possible. (You have pigeon in the pubspec, so I'm surprised it's not being used here.)

Reply:
Yes, we explored Pigeon but decided, later on, to keep consistent with the ´image_picker´. We'll modify our approach to use pigeons, shouldn't be too much work 😉.

Reply:
`image_picker` doesn't use Pigeon only because it predates Pigeon, and we haven't devoted time to migrating it yet.

Reply:
We made a refactor to use pigeons, let us know what you think!